tests/opcodes.cpp

```diff
@@ -0,0 +1,104 @@
+
```

Comment:
Maybe it makes sense to rename this file to the opcode you're testing? Like `opcode_add` or something like that

Reply:
TBH I figured that we'd throw all of those types of tests into a single file so we can centralize the helper methods. It it proves to be too unwieldy then we can split it up but I don't think it's necessary now.